contentcuration/contentcuration/views/base.py

```diff
@@ -352,3 +361,55 @@ def download_channel_content_csv(request, channel_id):
     generatechannelcsv_task.delay(channel_id, site.domain, request.user.id)
 
     return HttpResponse({"success": True})
+
+
+# Taken from kolibri.core.views which was
+# modified from django.views.i18n
+@require_POST
+def set_language(request):
+    """
+    Since this view changes how the user will see the rest of the site, it must
+    only be accessed as a POST request. If called as a GET request, it will
+    error.
+    """
+    payload = json.loads(request.body)
+    lang_code = payload.get(LANGUAGE_QUERY_PARAMETER)
+    next_url = urlsplit(payload.get("next")) if payload.get("next") else None
+    if lang_code and lang_code in SUPPORTED_LANGUAGES:
+        if next_url and is_valid_path(next_url.path):
+            # If it is a recognized path, then translate it to the new language and return it.
+            next_path = urlunsplit(
+                (
+                    next_url[0],
+                    next_url[1],
+                    translate_url(next_url[2], lang_code),
+                    next_url[3],
+                    next_url[4],
+                )
+            )
+        else:
+            # Handle this gracefully on the frontend
+            return HttpResponse(status=500)
```

Comment:
I don't really like us explicitly returning a 500, as it is indicative that something has gone very wrong.

I think best to just return a default URL ("/") and making the current base url part of the i18n prefixed URLs.